test/aqua/operators/test_op_construction.py

```diff
@@ -222,6 +247,23 @@ def test_matrix_to_instruction(self):
             with self.assertRaises(ExtensionError):
                 matop.to_instruction()
 
+    def test_matrix_to_gate(self):
+        """Test MatrixOp.to_gate yields a Gate object."""
+        matop = (H ^ 3).to_matrix_op()
+        with self.subTest('assert to_gate returns Gate'):
+            self.assertIsInstance(matop.to_gate(), Gate)
```

Comment:
Is there some check we can do on the gate to make sure that the conversion code in to_gate() is working beyond just checking its type. Can we check the value in some way - for instance doing to_matrix on the gate and assuring this is as expected too, or whatever seems reasonable to check that the gate is what we expect beyond just checking type.